deployer/deployer.py

```diff
@@ -297,23 +296,64 @@ def deploy_version_without_eureka(self, version, health_check, health_check_port
             }
         }
 
-        request = self.request("deployment/start", json.dumps(data))
-
-        time.sleep(30)
+        success = self.start_deployment(data, version)
+        if success:
+            self.resize_asg(asg_name=version)
 
-        self.resize_asg(asg_name=version)
+            good_deploy = self.wait_asg_ready(version=version,
+                                              health_check=health_check,
+                                              health_check_port=health_check_port)
 
-        good_deploy = self.wait_asg_ready(version=version,
-                                          health_check=health_check,
-                                          health_check_port=health_check_port)
+            if good_deploy:
+                if remove_old:
+                    self.remove_old_asg(new_asg_name=version)
+            else:
+                self.disable_asg(asg_name=version)
 
-        if good_deploy:
-            if remove_old:
-                self.remove_old_asg(new_asg_name=version)
         else:
-            self.disable_asg(asg_name=version)
+            raise Exception("Fucking asgard!")
+
+    def start_deployment(self, data, version):
```

Comment:
@Arcontus yes but this method doesn't only starts the deploy (not only executes this url) waits until it finish, so that the meaning of the name is different than the function does, I prefer something like "deploy_to_asgard".

This function only have sense if it is called inside the class so that must be private (I know that there are a lot of methods in this class that needs the same)